source/guides/core-concepts/introduction-to-cypress.md

```diff
@@ -385,6 +385,127 @@ it('changes the URL when "awesome" is clicked', function() {
 
 Cypress doesn't kick off the browser automation magic until the test function exits.
 
+### Mixing Async and Sync code
+
+Remembering that Cypress commands run asynchronously is important if you are attempting to mix Cypress commands with synchronous code. Synchronous code will execute immediately - not waiting for the Cypress commands above it to execute.
+
+**{% fa fa-warning red %} Incorrect Usage**
+
+In the example below, the `el` evaluates immediately, before the `cy.visit()` has executed, so will always evaluate to an empty array.
+
+```js
+it('does not work as we expect', () => {
+  cy.visit('/my/resource/path') // Nothing happens yet
+
+  cy.get('.awesome-selector')   // Still nothing happening
+    .click()                    // Nope, nothing
+
+  // Cypress.$ is synchronous
+  // this element only exists after you click the .awesome-selector
```

Comment:
I would explain this a little bit more
"There is no element to find yet - because the page "/my/resource/path" was only scheduled to be visited using `cy.visit()" above. So the iframe is empty - and you will get nothing back!
To really have the element, you should get the element AFTER the `cy.click` clicks on the button, which happens after `cy.visit` completes.

Reply:
Yah, oof, it’s hard to tersely explain this stuff 😅